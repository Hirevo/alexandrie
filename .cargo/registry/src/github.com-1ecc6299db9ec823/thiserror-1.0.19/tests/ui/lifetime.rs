use thiserror::Error;

#[derive(Error, Debug)]
#[error("error")]
struct Error<'a>(#[from] Inner<'a>);

#[derive(Error, Debug)]
#[error("{0}")]
struct Inner<'a>(&'a str);

fn main() -> Result<(), Error<'static>> {
    Err(Error(Inner("some text")))
}
