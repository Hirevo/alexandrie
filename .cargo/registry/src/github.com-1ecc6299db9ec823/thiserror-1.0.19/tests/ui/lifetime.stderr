error: non-static lifetimes are not allowed in the source of an error
 --> $DIR/lifetime.rs:5:18
  |
5 | struct Error<'a>(#[from] Inner<'a>);
  |                  ^^^^^^^^^^^^^^^^^
